### PR TITLE
AMaybe tests

### PR DESCRIPTION
**Pull request by edmundnoble**
Depends on #1963.

Also rephrases the `Maybe2` docs, and adds some extra instances.

### CODE REVIEW

**Comment by edmundnoble** [comment]
Marking as WIP because testing as a free pointed set isn't done and because I want `AMaybe2` tests in here too.

File: base/shared/src/main/scala/scalaz/data/acatenable1.scala [hunk 1]
```diff
@@ -54,14 +48,31 @@ sealed abstract class ACatenable1[=>:[_, _], A, B] {
...
+  /**
+   * Allows you to view the internal structure of an ACatenable1.
+   * Not stack-safe.
+   */
+  final def fold[G[_, _]](f: Forall3.Prototype[λ[(X, Y, Z) => (G[X, Y], G[Y, Z]) => G[X, Z]]])(z: =>: ~~> G): G[A, B] =
+    this match {
+      case Chain(a, b) => f.apply(a.fold[G](f)(z), b.fold[G](f)(z))
+      case Lift(a)     => z.apply(a)
+    }
+
```
Comment (emilypi) [inline base/shared/src/main/scala/scalaz/data/acatenable1.scala]: so dank

File: base/shared/src/main/scala/scalaz/tc/unfoldable1.scala [hunk 2]
```diff
@@ -1,4 +1,5 @@
-package scalaz.tc
+package scalaz
+package tc
 
 import scalaz.data.Maybe
```
Comment (emilypi) [inline base/shared/src/main/scala/scalaz/tc/unfoldable1.scala]: 👍

File: base/shared/src/main/scala/scalaz/tc/debug.scala [hunk 3]
```diff
@@ -83,6 +83,13 @@ object DebugClass {
     }
   }
 
+  implicit def tuple2Debug[A, B](implicit A: Debug[A], B: Debug[B]): Debug[(A, B)] = {
+    import Scalaz.debugInterpolator
+    DebugClass.instance[(A, B)] {
+      case (a, b) => z"($a, $b)"
```
Comment (emilypi) [inline base/shared/src/main/scala/scalaz/tc/debug.scala]: Can you give a quick rundown of what this interpolator does?

File: microsite/src/main/tut/data/Maybe2.md [hunk 4]
```diff
@@ -6,7 +6,11 @@ title:  "Maybe2"
 
 # Maybe2 [![GitHub](../img/github.png)](https://github.com/scalaz/scalaz/blob/series/8.0.x/base/shared/src/main/scala/scalaz/data/maybe2.scala)
 
-`Maybe2` is a specialized version of [Maybe](./Maybe.html) that avoids allocation of [Tuple2](https://www.scala-lang.org/api/current/scala/Tuple2.html) instances.
+`Maybe2[A, B]` is isomorphic to `Maybe[(A, B)]`.
+The only difference between `Maybe2[A, B]` and `Maybe[(A, B)]` is that the first takes less
+memory.
+
+`Maybe2` is used to implement `IList`; each cons cell is a `Maybe2`.
```
Comment (emilypi) [inline microsite/src/main/tut/data/Maybe2.md]: This is fine, but we probably need to expand this a little more for folks

File: tests/src/main/scala/scalaz/tests/ACatenable1Tests.scala [hunk 5]
```diff
@@ -19,8 +19,8 @@ object ACatenable1Tests {
   // these type aliases unnerve me with their presence
   type R[A, B] = ACatenable1[Map, A, B] => Map[A, B]
   type L[A, B] = Map[A, B] => ACatenable1[Map, A, B]
-  val foldF: ACatenable1[Map, ?, ?] ~~> Map = Forall2.of[R](_.fold)
-  val liftF: Map ~~> ACatenable1[Map, ?, ?] = Forall2.of[L](lift(_))
+  val foldBalancedF: ACatenable1[Map, ?, ?] ~~> Map = Forall2.of[R](_.foldBalanced)
```
Comment (emilypi) [inline tests/src/main/scala/scalaz/tests/ACatenable1Tests.scala]: Do we have binatural transformations documented? That should definitely be on the list as we're adding more than a few instances here.

File: base/shared/src/main/scala/scalaz/tc/debug.scala (hunk 3 above)
Comment (emilypi) [reply base/shared/src/main/scala/scalaz/tc/debug.scala]: nice!

**Pull request merged**